src/screens/AddressList.js

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { t } from 'ttag'
+import HathorPaginate from '../components/HathorPaginate';
+import HathorAlert from '../components/HathorAlert';
+import { EXPLORER_BASE_URL, WALLET_HISTORY_COUNT } from '../constants';
+import helpers from '../utils/helpers';
+import path from 'path';
+
+import hathorLib from '@hathor/wallet-lib';
+
+/**
+ * Screen that has a list of addresses of the wallet
+ *
+ * @memberof Screens
+ */
+class AddressList extends React.Component {
+  /**
+   * addresses {Array} All wallet addresses data {'address', 'index', 'numberOfTransactions'}
+   * filteredAddresses {Array} addresses state after search
+   * page: {Number} Current page of the list
+   * totalPages: {Number} Total number of pages of the list
+   * filtered: {Boolean} If the list is filtered
+   * transactionsByAddress: {Object} Object with number of transactions for each address
+   */
+  state = {
+    addresses: [],
+    filteredAddresses: [],
+    page: 1,
+    totalPages: 0,
+    filtered: false,
+    transactionsByAddress: null,
+  }
+
+  componentDidMount = () => {
+    const walletData = hathorLib.wallet.getWalletData();
+    const addressKeys = walletData.keys;
+    const addresses = [];
+    const transactionsByAddress = {};
+    for (const key in addressKeys) {
+      const addressData = {
+        address: key,
+        index: addressKeys[key].index,
+      };
+
+      addresses.push(addressData);
+      transactionsByAddress[key] = 0;
+    }
+
+    this.updateNumberOfTransactions(transactionsByAddress, walletData.historyTransactions);
+
+    this.setState({ addresses, filteredAddresses: addresses, totalPages: this.getTotalPages(addresses), transactionsByAddress });
+  }
+
+  /**
+   * Return total number of pages of the list
+   *
+   * @param {Array} array Array of addresses of the list
+   *
+   * @return {Number} Total number of pages of the list
+   */
+  getTotalPages = (array) => {
+    return Math.ceil(array.length / WALLET_HISTORY_COUNT);
+  }
+
+  /**
+   * Update number of transactions for each address
+   *
+   * @param {Object} addressData Transactions by address object {'address': 0}
```

Comment:
I think `{'address': 0}` may be misleading. I'd use `{[address]: 0}`.

Reply:
Changed